File: src/commands/targets.rs
use anyhow::{Context, Result};

use crate::client::{check_response, ApiClient};
use crate::models::Target;
use crate::output::{print_list, short_uuid, OutputMode};

pub async fn run(client: &ApiClient, mode: OutputMode) -> Result<()> {
    let resp = client
        .get("/v1/targets")
        .send()
        .await
        .context("Failed to list targets")?;
    let resp = check_response(resp).await?;
    let targets: Vec<Target> = resp.json().await.context("Failed to parse targets")?;

    print_list(
        mode,
        &targets,
        &["ID", "FAMILY", "DEVICE", "PACKAGE", "BOARD", "LANE"],
        |t| {
            vec![
                short_uuid(&t.id),
                t.family.clone(),
                t.device.clone(),
                t.package.clone().unwrap_or_else(|| "-".into()),
                t.board.clone().unwrap_or_else(|| "-".into()),
                t.toolchain_lane.clone(),
            ]
        },
    );

    Ok(())
}

File: lib/commands/targets.js
// Copyright 2026 Hemi Labs, Inc.
// SPDX-License-Identifier: GPL-3.0-only

import { printList, shortUuid } from "../output.js";

export async function targetsCommand(client, json) {
  const resp = await client.get("/v1/targets");
  const targets = await resp.json();

  printList(
    json,
    targets,
    ["ID", "FAMILY", "DEVICE", "PACKAGE", "BOARD", "LANE"],
    (t) => [
      shortUuid(t.id),
      t.family,
      t.device,
      t.package || "-",
      t.board || "-",
      t.toolchain_lane,
    ],
  );
}
